File: 251002/Samoilenko/discussion/src/main/java/com/example/discussion/service/mapper/NoteMapper.java
package com.example.discussion.service.mapper;

import com.example.discussion.dto.NoteRequestTo;
import com.example.discussion.dto.NoteResponseTo;
import com.example.discussion.model.Note;
import org.mapstruct.Mapper;
import org.mapstruct.MappingTarget;

@Mapper(componentModel = "spring")
public interface NoteMapper {
    Note toEntity(NoteRequestTo dto); // Преобразует DTO в сущность

    NoteResponseTo toDto(Note entity); // Преобразует сущность в DTO

    void updateEntityFromDto(NoteRequestTo dto, @MappingTarget Note entity);
}

File: 251002/Nikonorova/labs3-5/discussion/Mappers/NoteMapper.cs
using discussion.DTO.Request;
using discussion.Models;
using discussion.DTO.Response;
using Riok.Mapperly.Abstractions;

namespace discussion.Mappers
{
    [Mapper]
    public static partial class NoteMapper
    {
        public static partial Note ToEntity(this NoteRequestTo noteRequestTo);
        public static partial NoteResponseTo ToResponse(this Note note);
        public static partial IEnumerable<NoteResponseTo> ToResponse(this IEnumerable<Note> notes);

        public static Note UpdateEntity(this Note note, NoteRequestTo updateNoteRequestTo) => new()
        {
            Id = note.Id,
            ArticleId = updateNoteRequestTo.ArticleId,
            Content = updateNoteRequestTo?.Content,
            Country = note.Country
        };
    }
}
